### PR TITLE
NEXUS-8665: NEXUS-8666: Graceful content validation responses

### PR DESCRIPTION
**Pull request by cstamas**
Changes:
- NX3 should gracefully handle content validation failures
- groups should silently skip members throwing content validation failures

Issues
https://issues.sonatype.org/browse/NEXUS-8665
https://issues.sonatype.org/browse/NEXUS-8666

CI
http://bamboo.s/browse/NX3-OSSF329

### CODE REVIEW

**Comment by mrprescott** [comment]
Jason recently mentioned the idea of a handler to do this, which seems like a better idea. If there was a top-of-stack handler in each recipe doing this conversion, then we wouldn't need similar code between ViewServlet and GroupHandler (i.e. to catch IllegalOperationException | InvalidContentException e).  For groups, the original .isSuccessful() check would be enough, since the problem would have been mapped to an http response already.

**Comment by cstamas** [comment]
good idea! will close this one when back

thanks (phone),
~t~

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/ViewServlet.java [hunk 1]
```diff
@@ -129,6 +131,15 @@ protected void service(final HttpServletRequest httpRequest, final HttpServletRe
       log.info("Illegal operation {} {}: {}", httpRequest.getMethod(), httpRequest.getRequestURI(), e.getMessage());
       send(null, HttpResponses.badRequest(e.getMessage()), httpResponse);
     }
+    catch (InvalidContentException e) {
+      log.info("Invalid content {} {}: {}", httpRequest.getMethod(), httpRequest.getRequestURI(), e.getMessage());
```
Comment (jdillon) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/ViewServlet.java]: why logged as info?

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupHandler.java [hunk 2]
```diff
@@ -142,10 +149,15 @@ protected Response getFirst(final @Nonnull Request request,
...
 
-      final ViewFacet view = member.facet(ViewFacet.class);
-      final Response response = view.dispatch(request);
+      try {
+        final ViewFacet view = member.facet(ViewFacet.class);
+        final Response response = view.dispatch(request);
 
-      responses.put(member, response);
+        responses.put(member, response);
+      } catch (IllegalOperationException | InvalidContentException e) {
```
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupHandler.java]: watch your formatting plz

**Comment by cstamas** [comment]
@mrprescott @jdillon updated to use handler

**Comment by mrprescott** [comment]
+1

**Comment by cstamas** [comment]
merged as 805571215aec61ab7cc134f05e72672bfeb8e69f

**Pull request closed without merging**